From cbe3491707617847fcc599feb4dff39f9ff8d100 Mon Sep 17 00:00:00 2001
From: Gianfranco Costamagna <costamagnagianfranco@yahoo.it>
Date: Fri, 8 May 2026 14:17:57 +0200
Subject: [PATCH] signal-args.py: don't kill init system, leading to system
 reboot

Sending a kill to pid 1 is making the pbuilder containers reboot. Please fix the test, spawn a process for it, but don't play with init system PIDs
---
 test/apps/signal/signal-args.py | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/test/apps/signal/signal-args.py b/test/apps/signal/signal-args.py
index 2cb336e0..bd721731 100755
--- a/test/apps/signal/signal-args.py
+++ b/test/apps/signal/signal-args.py
@@ -24,13 +24,13 @@ do_test(["--help"], 1)
 do_test(["--invalid"], 1, response="Error: One of -a or -p must be used.\n")
 do_test(["-p"], 1, response="Error: -p argument given but process ID missing.\n")
 do_test(["-p", "0"], 1, response="Error: Process ID must be >0.\n")
-do_test(["-p", "1"], 1, response="Error: No signal given.\n")
+do_test(["-p", "2"], 1, response="Error: No signal given.\n")
 do_test(["-a"], 1, response="Error: No signal given.\n")
-do_test(["-p", "1", "invalid"], 1, response="Error: Unknown signal 'invalid'.\n")
-do_test(["-p", "1", "config-reload"], 0)
-do_test(["-p", "1", "log-rotate"], 0)
-do_test(["-p", "1", "shutdown"], 0)
-do_test(["-p", "1", "tree-print"], 0)
-do_test(["-p", "1", "xtreport"], 0)
+do_test(["-p", "2", "invalid"], 1, response="Error: Unknown signal 'invalid'.\n")
+do_test(["-p", "2", "config-reload"], 0)
+do_test(["-p", "2", "log-rotate"], 0)
+do_test(["-p", "2", "shutdown"], 0)
+do_test(["-p", "2", "tree-print"], 0)
+do_test(["-p", "2", "xtreport"], 0)
 do_test(["-a", "config-reload"], 0)
 exit(0)
